src/plopp/widgets/clip3d.py

```diff
@@ -0,0 +1,395 @@
+
+        self.outlines = [
+            p3.LineSegments(
+                geometry=p3.EdgesGeometry(
+                    p3.PlaneBufferGeometry(width=width, height=height)
+                ),
+                material=p3.LineBasicMaterial(color=color, linewidth=linewidth),
+            ),
+            p3.LineSegments(
+                geometry=p3.EdgesGeometry(
+                    p3.PlaneBufferGeometry(width=width, height=height)
+                ),
+                material=p3.LineBasicMaterial(color=color, linewidth=linewidth),
+            ),
+        ]
+        if self._direction == 'x':
+            for outline in self.outlines:
+                outline.rotateY(0.5 * np.pi)
+        if self._direction == 'y':
+            for outline in self.outlines:
+                outline.rotateX(0.5 * np.pi)
+
+        center = [var.mean().value for var in self._limits]
+        vmin = self._limits[axis][0].value
+        vmax = self._limits[axis][1].value
+        dx = vmax - vmin
+        delta = 0.05 * dx
+        self.slider = ipw.FloatRangeSlider(
+            min=vmin,
+            max=vmax,
+            value=[center[axis] - delta, center[axis] + delta],
+            step=dx * 0.01,
+            description=direction.upper(),
+            style={'description_width': 'initial'},
+            layout={'width': '470px', 'padding': '0px'},
+        )
+
+        self.cut_visible = ipw.Button(
+            icon='eye-slash',
+            tooltip='Hide cut',
+            layout={'width': '16px', 'padding': '0px'},
+        )
+
+        for outline, val in zip(self.outlines, self.slider.value):
+            pos = list(center)
+            pos[axis] = val
+            outline.position = pos
+
+        self.unit_label = ipw.Label(f'[{self._unit}]')
+        self.cut_visible.on_click(self.toggle)
+        self.slider.observe(self.move, names='value')
+
+        super().__init__([self.slider, ipw.Label(f'[{self._unit}]'), self.cut_visible])
+
+    def toggle(self, owner):
+        """
+        Toggle the visibility of the cut on and off.
+        """
+        self.visible = not self.visible
+        for outline in self.outlines:
+            outline.visible = self.visible
+        self.slider.disabled = not self.visible
+        owner.icon = 'eye-slash' if self.visible else 'eye'
+        owner.tooltip = 'Hide cut' if self.visible else 'Show cut'
+        self._plain_update()
+
+    def toggle_border(self, value: bool):
+        """
+        Toggle the border visibility.
+        """
+        for outline in self.outlines:
+            outline.visible = value
+
+    def move(self, value: Dict[str, Any]):
+        """
+        Move the outline of the cut according to new position given by the slider.
+        """
+        # Early return if relative difference between new and old value is small
+        old = np.array(value['old'])
+        if np.abs((np.array(value['new']) - old) / old).max() < 1e-6:
```

Comment:
What are `value['new']` and `value['old']` here?

Reply:
I'm thinking that if they are slider values and the interval is something like `[1000_000, 1000_001]` then this condition might always be true. But if they're always small then no problem. Why is the check needed? We're debouncing / throttling anyway right?

Reply:
This was serving a double purpose where it would also prevent flickering of an existing cut when adding a new one. I updated the comment and changed the threshold so that it is based on the magnitude of the step of the slider (which itself is based on the range of the slider).